### PR TITLE
BUG Caching CV indices to avoid calling cv.split twice

### LINKED ISSUE #16: `_base_est_fit_predict` looks wrong
I believe cv.split(X, y) might give different results when called at different times. There might be randomness involved like shuffling, etc.

This is problematic because `ybase`, which is based on the old `train` & `test` but not coincide with with the new `test` and therefore `y[test]`.

old `train` & `test`:

```
for train, test in cv.split(X, y):
    for name, est in self.estimator_list[:-1]:
        # adapted from sklearn.model_selection._fit_and_predict
        # Adjust length of sample weights
        fit_params_est_adjusted = dict([
            (k, _index_param_value(X, v, train))
            for k, v in fit_params_ests[name].items()])

        # Fit estimator on training set and score out-of-sample
        _jobs.append(delayed(_fit_predict)(
            clone(est),
            X[train],
            y[train],
            X[test],
            **fit_params_est_adjusted))
```

new `train` & `test`:

```
# Extract the results from joblib
Xmeta, ymeta = None, None
for train, test in cv.split(X, y):
    ybase = np.empty((y[test].shape[0], 0))
    for name, est in self.estimator_list[:-1]:
        # Build design matrix out of out-of-sample predictions
        ybase = np.hstack((ybase, _out.pop(0)))

    # Append the test outputs to what will eventually be the features
    # for the meta-estimator.
    if Xmeta is not None:
        ymeta = np.concatenate((ymeta, y[test]))
        Xmeta = np.vstack((Xmeta, ybase))
    else:
        Xmeta = ybase
        ymeta = y[test]
```

### PR DESCRIPTION
**Pull request by kcrum**
When `shuffle=True` and no random state is set, a cross-validation generator generates new indices each time `cv.split` is called. Since we need the exact same CV folds twice, we're now storing the indices. 

Closes #16.

### CODE REVIEW

**Comment by kcrum** [comment]
Oops, forgot to increment micro version and CHANGELOG. Fixing now...

**Comment by stephen-hoover** [comment]
@kcrum , version number increases should be in a separate PR.

File: civismlext/test/test_stacking.py [hunk 1]
```diff
@@ -729,6 +729,22 @@ def test_fit_regression():
...
 
 
+def test_cv_shuffle_indices():
+    """Make sure xmeta and ymeta retain the correct order, even when the CV
+    generator is shuffling.
+    """
+    estlist = [('be1', PassThruReg()),
+               ('be2', PassThruReg()),
+               ('meta', PassThruReg())]
+    sr = StackedRegressor(estlist, cv=KFold(n_splits=2, shuffle=True))
```
Comment (elsander) [inline civismlext/test/test_stacking.py]: Maybe I'm missing something, but I'm not sure why this test passes. You don't set a random seed in `cv`. Why is the shuffle order guaranteed to be consistent with `inds` here?

Comment (kcrum) [reply civismlext/test/test_stacking.py]: Ah, yes. The shuffle order doesn't need to be consistent with `inds`. All that matters is that the *relative* orders of `xmeta` and `ymeta` are wapped with respect to each other.

Comment (elsander) [reply civismlext/test/test_stacking.py]: Ok, I see what you're saying. I think this is testing the right thing, but I think it needs a documentation block to explain why. I would explain what you told me offline about two-fold cross-validation meaning that the train and test blocks will be of length 3 for each fold, and that the test block will be all indices not in the training block, and that that is what `inds` represents here.

I tried to think of a more intuitive test, but none of the problem parts of the function are stored in attributes, so we can't really test against them. I think this test will be fine if it has a big explanatory comment block somewhere.
